gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestPlanMapperTest.java

```diff
@@ -120,212 +198,495 @@ public void When_MappingAuthoriseResourceWithAllOptionals_Expect_AllFieldsToBeMa
-
-        var medicationRequest =
-            medicationRequestPlanMapper.mapToPlanMedicationRequest(ehrExtract, medicationStatement.get(), supplyAuthorise.orElseThrow(),
-                PRACTISE_CODE);
+    public void When_MappingDiscontinueWithPertinentInformation_Expect_StatusReasonAdded() {
+        var medicationStatementXml = """
+            <MedicationStatement xmlns="urn:hl7-org:v3" classCode="SBADM" moodCode="INT">
+                <id root="B4D70A6D-2EE4-41B6-B1FB-F9F0AD84C503"/>
+                <component typeCode="COMP">
+                    <ehrSupplyAuthorise classCode="SPLY" moodCode="INT">
+                        <id root="TEST_ID"/>
+                    </ehrSupplyAuthorise>
+                </component>
+                <component typeCode="COMP">
+                    <ehrSupplyDiscontinue classCode="SPLY" moodCode="RQO">
+                        <id root="D0BF39CA-E656-4322-879F-83EE6E688053"/>
+                        <availabilityTime value="20060426"/>
+                        <reversalOf typeCode="REV">
+                            <priorMedicationRef classCode="SBADM" moodCode="ORD">
+                                <id root="TEST_ID"/>
+                            </priorMedicationRef>
+                        </reversalOf>
+                        <pertinentInformation typeCode="PERT">
+                            <pertinentSupplyAnnotation classCode="OBS" moodCode="EVN">
+                                <text>Patient no longer requires these</text>
+                            </pertinentSupplyAnnotation>
+                        </pertinentInformation>
+                    </ehrSupplyDiscontinue>
+                </component>
+            </MedicationStatement>
+            """;
+        var ehrExtract = unmarshallEhrExtractFromMedicationRequestXml(medicationStatementXml);
+        var medicationStatement = extractMedicationStatement(ehrExtract);
+        var supplyAuthorise = extractSupplyAuthorise(medicationStatement);
 
+        var medicationRequest = medicationRequestPlanMapper.mapToPlanMedicationRequest(
+            ehrExtract,
+            medicationStatement,
+            supplyAuthorise,
+            PRACTISE_CODE
+        );
         var statusExt = medicationRequest.getExtensionsByUrl(MEDICATION_STATUS_REASON_URL);
-        assertThat(statusExt).hasSize(1);
-
-        var statusReasonExt = statusExt.get(0).getExtensionsByUrl("statusReason");
-        assertThat(statusReasonExt).hasSize(1);
-
+        var statusReasonExt = statusExt.get(0).getExtensionsByUrl(STATUS_REASON);
         var statusReason = (CodeableConcept) statusReasonExt.get(0).getValue();
 
-        assertThat(statusReason.getText()).isEqualTo("Patient no longer requires these");
+        assertAll(
+            () -> assertThat(statusExt).hasSize(1),
+            () -> assertThat(statusReasonExt).hasSize(1),
+            () -> assertThat(statusReason.getText()).isEqualTo("Patient no longer requires these")
+        );
```

Comment:
I wonder if these assertions will sometimes be less clear now, let's say that `statusExt` has a size of 0, it would now be a NullReferenceException on line 303 🤔 .